0001-changelog-format.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: Standarize on the format of changelogs
+- Start Date: 2019-02-15
+- RFC Status: Final Comment
+- RFC PR: http://github.com/saltstack/salt-enhancement-proposals/pull/2 
+
+# Summary
+[summary]: #summary
+
+This RFC proposes a standard changelog format for software released by the SaltStack project.
+
+# Motivation
+[motivation]: #motivation
+
+The current format for changelog entries at Salt is unstructured and can be difficult to read.
+
+Many users of Salt have complained that they weren't aware of a breaking change which may
+be the result of a poorly structured and inconsistent changelog between releases.
+
+# Design
+[design]: #detailed-design
+
+This RFC standardizes Salt on the 1.0.0 spec of project changelogs outlined at
+https://keepachangelog.com/en/1.0.0/
+
+Specifically, it provides for the following principles:
+
+- Changelogs are for humans, not machines.
+- There should be an entry for every single version.
+- The same types of changes should be grouped.
+- Versions and sections should be linkable.
+- The latest version comes first.
+- The release date of each version is displayed.
+- Mention whether you follow Semantic Versioning.
+
+Types of changes
+----------------
+- Added for new features.
+- Changed for changes in existing functionality.
+- Deprecated for soon-to-be removed features.
```

Comment:
Since a critical area most users look for is "Are there any breaking changes in this release", would it make sense to incorporate "Back-ward incompatible functionality" in separate section ? or would that be part of any of the sections listed.

Reply:
I'm not sure how "backward-incompatible" and "deprecated" are different. Could you clarify?

Reply:
Backward-incompatible to me means how you use a service/functionality, or interface with a module has changed but the overall service continues to be offered. For eg.,
https://docs.saltstack.com/en/develop/topics/releases/2019.2.0.html#non-backward-compatible-change-to-yaml-renderer 

Deprecation indicates a service/functionality/something is 'discontinued' now or in future releases. Like no support for Py2 after its EOL. 

Both necessiates user actions to migrate, so I can see why both are similar. 

Since you are well versed in industry notations @cachedout,  I will leave this to you :)

Reply:
This makes sense and I agree with you. I can see how it would make sense to distinguish between deprecation notices which indicate that something *will* be removed/changed and backward compatibility indicates whether the behavior *has* been changed.

Reply:
What if we do something like this?

---

## Changes

- **Breaking** - Non-backward compatible change to YAML renderer
- **Breaking** - Frobnosticated the frizzle sticks
- Whirlygigs now have 20% more whirl
- Gew gaws have 15% less gaw, and 30% more gew